core-web/src/main/java/org/silverpeas/core/webapi/media/streaming/StreamingProviderDataEntity.java

```diff
@@ -0,0 +1,247 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.silverpeas.core.webapi.media.streaming;
+
+import org.silverpeas.core.io.media.Definition;
+import org.silverpeas.core.util.JSONCodec;
+import org.silverpeas.core.util.URLUtil;
+import org.silverpeas.core.util.media.streaming.StreamingProvider;
+import org.silverpeas.core.util.media.streaming.StreamingProvidersRegistry;
+import org.silverpeas.core.web.rs.WebEntity;
+import org.silverpeas.core.webapi.media.MediaDefinitionEntity;
+
+import javax.ws.rs.core.UriBuilder;
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+import javax.xml.bind.annotation.XmlTransient;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.silverpeas.core.util.StringUtil.isDefined;
+import static org.silverpeas.core.webapi.media.streaming.StreamingRequester.getJsonOembedAsString;
+
+/**
+ * This entity ensures that all streaming data are formatted in a single way whatever the
+ * streaming provider.
+ * @author silveryocha
+ */
+@XmlRootElement
+@XmlAccessorType(XmlAccessType.FIELD)
+public class StreamingProviderDataEntity implements WebEntity {
+  private static final long serialVersionUID = 4017230238128160967L;
+
+  @XmlElement(defaultValue = "")
+  private URI uri;
+
+  @XmlTransient
+  private String originalUrl;
+
+  @XmlElement
+  private String provider;
+
+  @XmlElement
+  private String title;
+
+  @XmlElement
+  private String author;
+
+  @XmlElement
+  private String formattedDurationHMS;
+
+  @XmlElement
+  private MediaDefinitionEntity definition =
+      MediaDefinitionEntity.createFrom(Definition.fromZero());
+
+  @XmlElement
+  private String embedHtml;
+
+  @XmlElement
+  private URI thumbnailUrl;
+
+  @XmlElement
+  private MediaDefinitionEntity thumbnailDefinition;
+
+  @XmlElement
+  private List<URI> thumbnailPreviewUrls = new ArrayList<>();
+
+  /**
+   * Creates a streaming provider data entity from specified homepage url.
+   * @param homepageUrl the streaming home page url.
+   * @return the streaming provider data entity representing the specified streaming.
+   */
+  public static Optional<StreamingProviderDataEntity> from(final String homepageUrl) {
+    return StreamingProvidersRegistry.get().getFromUrl(homepageUrl)
+        .map(p -> {
+          final OembedDataEntity oembedData =
+              JSONCodec.decode(getJsonOembedAsString(homepageUrl), OembedDataEntity.class);
+          final StreamingProviderDataEntity entity;
+          if ("youtube".equals(p.getName())) {
```

Comment:
I propose to define "youtube" or "vimeo" as a constant defined in their respective data entity (YoutubeDataEntity and VimeoDataEntity). A factory of StreamingProvidedDataEntity can be used here to construct the correct instance of such an entity.

Reply:
OK, I will implement factory.